src/Sparrow.Server/Platform/RavenProcess.cs

```diff
@@ -0,0 +1,209 @@
+                handler?.Invoke(this, line);
+            }
+        }
+
+        public delegate void StreamWriteDelegate(Span<byte> bytes);
+
+        private readonly Logger _logger = LoggingSource.Instance.GetLogger<RavenProcess>("RavenProcess");
+        private IntPtr _pid = IntPtr.Zero;
+        private SafeFileHandle StandardOutAndErr { get; set; }
+        private SafeFileHandle StandardIn { get; set; }
+
+        public void Start(CancellationToken ctk)
+        {
+            if (StartInfo?.FileName == null)
+                throw new InvalidOperationException("RavenProcess Start() must be supplied with valid startInfo object and set Filename");
+
+            var rc = Pal.rvn_spawn_process(StartInfo.FileName, StartInfo.Arguments, out var pid, out var stdin, out var stdout, out var errorCode);
+            if (rc != PalFlags.FailCodes.Success)
+                PalHelper.ThrowLastError(rc, errorCode, $"Failed to spawn command '{StartInfo.FileName} {StartInfo.Arguments}'");
+
+            _pid = pid;
+            StandardOutAndErr = stdout;
+            StandardIn = stdin;
+        }
+
+        public static void Start(string filename, string arguments, StreamWriteDelegate outputDel)
+        {
+            using (var ravenProcess = new RavenProcess {StartInfo = new ProcessStartInfo {FileName = filename, Arguments = arguments}})
+            {
+                ravenProcess.Start(CancellationToken.None);
+                ravenProcess.ReadTo(outputDel);
+            }
+        }
+
+        private void ReadTo(StreamWriteDelegate outputDel)
+        {
+            var bytes = new byte[4096];
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            {
+                var read = fs.Read(bytes, 0, 4096);
+                while (read != 0)
+                {
+                    outputDel?.Invoke(new Span<byte>(bytes, 0, read));
+                    read = fs.Read(bytes, 0, 4096);
+                }
+            }
+        }
+
+        public Task<string> ReadToEndAsync()
+        {
+            using (var fs = new FileStream(StandardOutAndErr, FileAccess.Read))
+            using (var sr = new StreamReader(fs))
+            {
+                return sr.ReadToEndAsync();
+            }
+        }
+
+        private void Kill()
+        {
+            if (_pid != IntPtr.Zero)
+            {
+                var rc = Pal.rvn_kill_process(_pid, out var errorCode);
+                if (rc != PalFlags.FailCodes.Success)
+                    PalHelper.ThrowLastError(rc, errorCode, $"Failed to kill proc id={_pid.ToInt64()}. Command: '{StartInfo.FileName} {StartInfo.Arguments}'");
+            }
+        }
+
+        private void ReadLines(StreamReader sr, Action<object, string> lineOutputHandler, CancellationToken ctk)
+        {
+            while (ctk.IsCancellationRequested == false)
+            {
+                var lineTask = sr.ReadLineAsync();
+                if (lineTask.Wait(-1, ctk) == false)
```

Comment:
Minor: you can use `Timeout.Infinite` instead of `-1` to make it a bit more clear